intrahospital_api/apis/prod_api.py

```diff
@@ -811,9 +811,14 @@ def results_for_hospital_number(self, hospital_number):
         returns all the results for an MRN
         aggregated into labtest: observations([])
         """
-        other_hns = self.query_for_zero_prefixed(hospital_number)
+        zero_prefixed_mrns = self.query_for_zero_prefixed(hospital_number)
         raw_rows = self.raw_data(hospital_number)
-        for other_hn in other_hns:
-            raw_rows.extend(self.raw_data(other_hn))
+        for zero_prefixed_mrn in zero_prefixed_mrns:
+            raw_rows.extend(self.raw_data(zero_prefixed_mrn))
+        merged_mrns = list(set(MergedMRN.objects.filter(
+            patient__demographics__hospital_number=hospital_number
+        ).values_list('mrn', flat=True)))
+        for merged_mrn in merged_mrns:
+            raw_rows.extend(self.raw_data(merged_mrn))
```

Comment:
Why can't merged MRNs have 0-prefixed lab test identifier entries?

Reply:
The cerner master file implicitly.

So the logic we're using is. A patient is merged if they exist in the cerner master file as inactive and have a merge to another patient. As the cerner master file contains no zero prefixes, a merged MRN can't have a zero prefix.

Reply:
actually after conversation. Good call, done.